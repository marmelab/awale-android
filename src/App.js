// import React from 'react';
// // mport Navigation from './Navigation';
// import { Text } from 'react-native';
//
// const App = () => (
//     <Text>oui</Text>
// );
//
// export default App;
