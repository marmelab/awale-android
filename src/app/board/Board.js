import React, { PropTypes, Component } from 'react';
import { StyleSheet, View } from 'react-native';
import PitButton from './PitButton';

const styles = StyleSheet.create({
    board: {
        flexDirection: 'row',
        flexWrap: 'wrap',
        backgroundColor: '#f39c12',
        justifyContent: 'space-around',
        borderRadius: 10,
        height: 130,
        padding: 10,
        width: 340,
    },
});

export default class Board extends Component {
    static propTypes = {
        board: PropTypes.arrayOf(React.PropTypes.number).isRequired,
        pickPebble: PropTypes.func.isRequired,
        currentIndexPlayer: PropTypes.number.isRequired,
        canPlay: PropTypes.bool,
    }

    static defaultProps = {
        canPlay: true,
    }

    pickPebble = (position) => {
        this.props.pickPebble(position);
    }

    render() {
        const size = this.props.board.length;
        const halfSize = size / 2;
        const topBoard = this.props.board.slice(halfSize, size).reverse();
        const bottomBoard = this.props.board.slice(0, halfSize);

        return (
            <View style={styles.board}>
                {topBoard.map((pit, i) =>
                    <PitButton
                        onPress={this.pickPebble}
                        pitValue={pit}
                        pitIndex={size - 1 - i}
                        enabled={this.props.currentIndexPlayer === 1 && this.props.canPlay}
                        key={`topBoard-${i}`}
                    />
                )}
                {bottomBoard.map((pit, i) =>
                    <PitButton
                        onPress={this.pickPebble}
                        pitValue={pit}
                        pitIndex={i}
                        enabled={this.props.currentIndexPlayer === 0}
                        key={`bottomBoard-${i}`}
                    />
                )}
            </View>
        );
    }
}
