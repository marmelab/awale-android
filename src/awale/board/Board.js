import React, { PropTypes, Component } from 'react';
import { StyleSheet, Text, View } from 'react-native';

const styles = StyleSheet.create({
    board: {
        flexDirection: 'row',
        flexWrap: 'wrap',
        backgroundColor: 'steelblue',
        justifyContent: 'space-around',
        height: 150,
        padding: 20,
        width: 350,
    },
    pit: {
        borderRadius: 20,
        height: 40,
        width: 40,
        marginLeft: 10,
        marginTop: 10,
        backgroundColor: 'red',
        textAlign: 'center',
        textAlignVertical: 'center',
    },
});

export default class Board extends Component {
    render() {
        const size = this.props.board.length;
        const halfSize = size / 2;
        const topBoard = this.props.board.slice(0, halfSize).reverse();
        const bottomBoard = this.props.board.slice(halfSize, size);

        return (
            <View style={styles.board}>
                {topBoard.map((pit, x) =>
                    <Text style={styles.pit} key={`topBoard-${x}`}>{pit}</Text>
                )}
                {bottomBoard.map((pit, x) =>
                    <Text style={styles.pit} key={`bottomBoard-${x}`}>{pit}</Text>
                )}
            </View>
        );
    }
}

Board.propTypes = {
    board: PropTypes.arrayOf(React.PropTypes.number).isRequired,
};
