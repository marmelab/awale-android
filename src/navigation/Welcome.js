// import React, { PropTypes, Component } from 'react';
// import { View, Text, Navigator, Button } from 'react-native';
//
// export default class Welcome extends Component {
//
//     startPlaying(playAgainstComputer) {
//         this.props.navigator.replace({ id: 'Play', againstComputer: playAgainstComputer });
//     }
//
//     render() {
//         return (
//             <View>
//                 <View >
//                     <Text>Awale</Text>
//                 </View>
//                 <View >
//                     <View style={{ marginBottom: 30 }}>
//                         <Button title="Play against computer" onPress={() => this.startPlaying(true)} />
//                     </View>
//                     <View style={{ marginBottom: 30 }}>
//                         <Button title="Play against another player" onPress={() => this.startPlaying(false)} />
//                     </View>
//                 </View>
//             </View>
//         );
//     }
// }
//
// Welcome.propTypes = {
//     navigator: PropTypes.instanceOf(Navigator).isRequired,
// };
