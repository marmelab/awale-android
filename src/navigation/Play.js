import React, { PropTypes, Component } from 'react';
import { View, Navigator, Button, Alert, StyleSheet } from 'react-native';

import { GAME_CONTINUE } from '../awale/constants/Constants';
import Board from '../app/board/Board';
import ScoreCircle from '../app/score/ScoreCircle';
import createPlayer from '../awale/player/Player';
import {
    create as createGame,
    playTurn,
    getCurrentPlayer,
    checkWinner,
} from '../awale/game/Game';
import { canPlayerPlayPosition } from '../awale/board/Board';

import config from '../../config';

const styles = StyleSheet.create({
    view: {
        flex: 1,
        backgroundColor: '#1abc9c',
        paddingLeft: 10,
        paddingRight: 10,
    },
    header: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
        marginBottom: 5,
    },
    boardContainer: {
        flex: 4,
    },
    scoreContainer: {
        flex: 2,
    },
});

export default class Play extends Component {
    static propTypes = {
        navigator: PropTypes.instanceOf(Navigator).isRequired,
        againstComputer: PropTypes.bool.isRequired,
    }

    constructor(props) {
        super(props);

        this.state = {
            game: this.getGameModel(),
            canPlay: true,
        };
    }

    getGameModel() {
        return createGame([createPlayer(0), createPlayer(1, !this.props.againstComputer)]);
    }

    handleGoHomeClick = () => {
        this.props.navigator.replace({ id: 'Welcome' });
    }

    handlePlayAgainClick = () => {
        this.setState({ game: this.getGameModel() });
    }

    showGameStatus = (gameState) => {
        const message = `Player ${gameState + 1} wins !`;
        Alert.alert(
            'Game winner',
            message,
            [
                {
                    text: 'Back to home ',
                    style: 'cancel',
                    onPress: this.handleGoHomeClick,
                },
                {
                    text: 'Play again',
                    style: 'default',
                    onPress: this.handlePlayAgainClick,
                },
            ],
            {
                cancelable: false,
            },
         );
    }

    pickPebble = (position) => {
        const player = getCurrentPlayer(this.state.game);
        const canPlay = canPlayerPlayPosition(player, this.state.game.board, position);
        if (!canPlay) {
            return;
        }

        let nextGame = playTurn(this.state.game, position);
        this.setState({ game: nextGame });

        nextGame = checkWinner(nextGame);
        if (nextGame.gameState !== GAME_CONTINUE) {
            this.showGameStatus(nextGame.gameState);
        } else {
            this.checkComputerTurn(nextGame);
        }
    }

    checkComputerTurn = (game) => {
        const player = getCurrentPlayer(game);
        if (player.isHuman || !this.props.againstComputer) {
            return;
        }

        this.setState({ canPlay: false });
        this.fetchColumn(game).then((bestPosition) => {
            this.pickPebble(bestPosition);
            this.setState({ canPlay: true });
        });
    }

    fetchColumn = (game) => {
        return fetch(config.apiUrl, {
            method: 'POST',
            body: JSON.stringify({ Score: game.score, Board: game.board }),
        })
        .then(response => response.text())
        .then(parseInt);
    }

    render() {
        const { game, canPlay } = this.state;
        const highlightPlayerOne = (game.currentIndexPlayer === 0);

        return (
            <View style={styles.view}>
                <View style={styles.header}>
                    <Button onPress={this.handleGoHomeClick} title="Back to home" color="#138a72" />
                </View>

                <View style={styles.scoreContainer}>
                    <ScoreCircle score={game.score[1]} highlight={!highlightPlayerOne} flexDirection="row-reverse" colorSyle="#34495e" />
                </View>

                <View style={styles.boardContainer}>
                    <Board
                        board={game.board}
                        currentIndexPlayer={game.currentIndexPlayer}
                        pickPebble={this.pickPebble}
                        canPlay={canPlay}
                    />
                </View>

                <View style={styles.scoreContainer}>
                    <ScoreCircle score={game.score[0]} highlight={highlightPlayerOne} flexDirection="row" colorSyle="#9b59b6" />
                </View>

            </View>
        );
    }
}
