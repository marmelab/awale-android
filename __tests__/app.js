import React from 'react';
import renderer from 'react-test-renderer';
import { Navigator } from 'react-native';
import App from '../src/App';
import Board from '../src/awale/board/Board';
import Welcome from '../src/navigation/Welcome';
import Play from '../src/navigation/Play';

it('renders the App component', () => {
    const tree = renderer.create(
        <App />
    ).toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders the Board component', () => {
    const cells = Array(12).fill(4);
    const tree = renderer.create(
        <Board board={cells} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders the Welcome component', () => {
    const navigator = new Navigator({
        initialRoute: { id: 'Welcome' },
        configureScene: (route, routeStack) => Navigator.SceneConfigs.FloatFromRight,
    });

    const tree = renderer.create(
        <Welcome navigator={navigator} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders the Play component', () => {
    const navigator = new Navigator({
        initialRoute: { id: 'Play' },
        configureScene: (route, routeStack) => Navigator.SceneConfigs.FloatFromRight,
    });

    const tree = renderer.create(
        <Play navigator={navigator} againstComputer />
    ).toJSON();
    expect(tree).toMatchSnapshot();
});
