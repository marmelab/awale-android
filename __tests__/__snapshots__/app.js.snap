// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the App component 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "overflow": "hidden",
      },
      undefined,
    ]
  }
>
  <View
    onMoveShouldSetResponder={[Function]}
    onMoveShouldSetResponderCapture={[Function]}
    onResponderEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderStart={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    onTouchStart={[Function]}
    style={
      Object {
        "backgroundColor": "transparent",
        "flex": 1,
        "overflow": "hidden",
      }
    }
  >
    <View
      onStartShouldSetResponderCapture={[Function]}
      pointerEvents="auto"
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "overflow": "hidden",
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "transform": Array [
              Object {
                "translateX": 0,
              },
              Object {
                "translateY": 0,
              },
              Object {
                "scaleX": 1,
              },
              Object {
                "scaleY": 1,
              },
              Object {
                "rotate": "0deg",
              },
              Object {
                "skewX": "0deg",
              },
              Object {
                "skewY": "0deg",
              },
            ],
          },
          null,
        ]
      }
    >
      <View>
        <View>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          >
            Awale
          </Text>
        </View>
        <View>
          <View
            style={
              Object {
                "marginBottom": 30,
              }
            }
          >
            <View
              accessibilityComponentType="button"
              accessibilityLabel={undefined}
              accessibilityTraits={
                Array [
                  "button",
                ]
              }
              accessible={true}
              hitSlop={undefined}
              isTVSelectable={true}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
              testID={undefined}
              tvParallaxProperties={undefined}
            >
              <View
                style={
                  Array [
                    Object {},
                  ]
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "#0C42FD",
                        "fontSize": 18,
                        "padding": 8,
                        "textAlign": "center",
                      },
                    ]
                  }
                >
                  Play against computer
                </Text>
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "marginBottom": 30,
              }
            }
          >
            <View
              accessibilityComponentType="button"
              accessibilityLabel={undefined}
              accessibilityTraits={
                Array [
                  "button",
                ]
              }
              accessible={true}
              hitSlop={undefined}
              isTVSelectable={true}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
              testID={undefined}
              tvParallaxProperties={undefined}
            >
              <View
                style={
                  Array [
                    Object {},
                  ]
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "#0C42FD",
                        "fontSize": 18,
                        "padding": 8,
                        "textAlign": "center",
                      },
                    ]
                  }
                >
                  Play against another player
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`renders the Board component 1`] = `
<View
  style={
    Object {
      "backgroundColor": "steelblue",
      "flexDirection": "row",
      "flexWrap": "wrap",
      "height": 150,
      "justifyContent": "space-around",
      "padding": 20,
      "width": 350,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "backgroundColor": "red",
          "borderRadius": 20,
          "height": 40,
          "marginLeft": 10,
          "marginTop": 10,
          "textAlign": "center",
          "textAlignVertical": "center",
          "width": 40,
        },
        Object {
          "backgroundColor": "orange",
        },
      ]
    }
  >
    4
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "backgroundColor": "red",
          "borderRadius": 20,
          "height": 40,
          "marginLeft": 10,
          "marginTop": 10,
          "textAlign": "center",
          "textAlignVertical": "center",
          "width": 40,
        },
        Object {
          "backgroundColor": "orange",
        },
      ]
    }
  >
    4
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "backgroundColor": "red",
          "borderRadius": 20,
          "height": 40,
          "marginLeft": 10,
          "marginTop": 10,
          "textAlign": "center",
          "textAlignVertical": "center",
          "width": 40,
        },
        Object {
          "backgroundColor": "orange",
        },
      ]
    }
  >
    4
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "backgroundColor": "red",
          "borderRadius": 20,
          "height": 40,
          "marginLeft": 10,
          "marginTop": 10,
          "textAlign": "center",
          "textAlignVertical": "center",
          "width": 40,
        },
        Object {
          "backgroundColor": "orange",
        },
      ]
    }
  >
    4
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "backgroundColor": "red",
          "borderRadius": 20,
          "height": 40,
          "marginLeft": 10,
          "marginTop": 10,
          "textAlign": "center",
          "textAlignVertical": "center",
          "width": 40,
        },
        Object {
          "backgroundColor": "orange",
        },
      ]
    }
  >
    4
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "backgroundColor": "red",
          "borderRadius": 20,
          "height": 40,
          "marginLeft": 10,
          "marginTop": 10,
          "textAlign": "center",
          "textAlignVertical": "center",
          "width": 40,
        },
        Object {
          "backgroundColor": "orange",
        },
      ]
    }
  >
    4
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    onPress={[Function]}
    style={
      Object {
        "backgroundColor": "red",
        "borderRadius": 20,
        "height": 40,
        "marginLeft": 10,
        "marginTop": 10,
        "textAlign": "center",
        "textAlignVertical": "center",
        "width": 40,
      }
    }
  >
    4
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    onPress={[Function]}
    style={
      Object {
        "backgroundColor": "red",
        "borderRadius": 20,
        "height": 40,
        "marginLeft": 10,
        "marginTop": 10,
        "textAlign": "center",
        "textAlignVertical": "center",
        "width": 40,
      }
    }
  >
    4
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    onPress={[Function]}
    style={
      Object {
        "backgroundColor": "red",
        "borderRadius": 20,
        "height": 40,
        "marginLeft": 10,
        "marginTop": 10,
        "textAlign": "center",
        "textAlignVertical": "center",
        "width": 40,
      }
    }
  >
    4
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    onPress={[Function]}
    style={
      Object {
        "backgroundColor": "red",
        "borderRadius": 20,
        "height": 40,
        "marginLeft": 10,
        "marginTop": 10,
        "textAlign": "center",
        "textAlignVertical": "center",
        "width": 40,
      }
    }
  >
    4
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    onPress={[Function]}
    style={
      Object {
        "backgroundColor": "red",
        "borderRadius": 20,
        "height": 40,
        "marginLeft": 10,
        "marginTop": 10,
        "textAlign": "center",
        "textAlignVertical": "center",
        "width": 40,
      }
    }
  >
    4
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    onPress={[Function]}
    style={
      Object {
        "backgroundColor": "red",
        "borderRadius": 20,
        "height": 40,
        "marginLeft": 10,
        "marginTop": 10,
        "textAlign": "center",
        "textAlignVertical": "center",
        "width": 40,
      }
    }
  >
    4
  </Text>
</View>
`;

exports[`renders the Play component 1`] = `
<View>
  <View>
    <View
      accessibilityComponentType="button"
      accessibilityLabel={undefined}
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#0C42FD",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Back to home
        </Text>
      </View>
    </View>
  </View>
  <View>
    <View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={Object {}}
      >
        Score : 
        0
      </Text>
    </View>
  </View>
  <View>
    <View
      style={
        Object {
          "backgroundColor": "steelblue",
          "flexDirection": "row",
          "flexWrap": "wrap",
          "height": 150,
          "justifyContent": "space-around",
          "padding": 20,
          "width": 350,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "backgroundColor": "red",
              "borderRadius": 20,
              "height": 40,
              "marginLeft": 10,
              "marginTop": 10,
              "textAlign": "center",
              "textAlignVertical": "center",
              "width": 40,
            },
            Object {
              "backgroundColor": "orange",
            },
          ]
        }
      >
        4
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "backgroundColor": "red",
              "borderRadius": 20,
              "height": 40,
              "marginLeft": 10,
              "marginTop": 10,
              "textAlign": "center",
              "textAlignVertical": "center",
              "width": 40,
            },
            Object {
              "backgroundColor": "orange",
            },
          ]
        }
      >
        4
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "backgroundColor": "red",
              "borderRadius": 20,
              "height": 40,
              "marginLeft": 10,
              "marginTop": 10,
              "textAlign": "center",
              "textAlignVertical": "center",
              "width": 40,
            },
            Object {
              "backgroundColor": "orange",
            },
          ]
        }
      >
        4
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "backgroundColor": "red",
              "borderRadius": 20,
              "height": 40,
              "marginLeft": 10,
              "marginTop": 10,
              "textAlign": "center",
              "textAlignVertical": "center",
              "width": 40,
            },
            Object {
              "backgroundColor": "orange",
            },
          ]
        }
      >
        4
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "backgroundColor": "red",
              "borderRadius": 20,
              "height": 40,
              "marginLeft": 10,
              "marginTop": 10,
              "textAlign": "center",
              "textAlignVertical": "center",
              "width": 40,
            },
            Object {
              "backgroundColor": "orange",
            },
          ]
        }
      >
        4
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "backgroundColor": "red",
              "borderRadius": 20,
              "height": 40,
              "marginLeft": 10,
              "marginTop": 10,
              "textAlign": "center",
              "textAlignVertical": "center",
              "width": 40,
            },
            Object {
              "backgroundColor": "orange",
            },
          ]
        }
      >
        4
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        onPress={[Function]}
        style={
          Object {
            "backgroundColor": "red",
            "borderRadius": 20,
            "height": 40,
            "marginLeft": 10,
            "marginTop": 10,
            "textAlign": "center",
            "textAlignVertical": "center",
            "width": 40,
          }
        }
      >
        4
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        onPress={[Function]}
        style={
          Object {
            "backgroundColor": "red",
            "borderRadius": 20,
            "height": 40,
            "marginLeft": 10,
            "marginTop": 10,
            "textAlign": "center",
            "textAlignVertical": "center",
            "width": 40,
          }
        }
      >
        4
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        onPress={[Function]}
        style={
          Object {
            "backgroundColor": "red",
            "borderRadius": 20,
            "height": 40,
            "marginLeft": 10,
            "marginTop": 10,
            "textAlign": "center",
            "textAlignVertical": "center",
            "width": 40,
          }
        }
      >
        4
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        onPress={[Function]}
        style={
          Object {
            "backgroundColor": "red",
            "borderRadius": 20,
            "height": 40,
            "marginLeft": 10,
            "marginTop": 10,
            "textAlign": "center",
            "textAlignVertical": "center",
            "width": 40,
          }
        }
      >
        4
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        onPress={[Function]}
        style={
          Object {
            "backgroundColor": "red",
            "borderRadius": 20,
            "height": 40,
            "marginLeft": 10,
            "marginTop": 10,
            "textAlign": "center",
            "textAlignVertical": "center",
            "width": 40,
          }
        }
      >
        4
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        onPress={[Function]}
        style={
          Object {
            "backgroundColor": "red",
            "borderRadius": 20,
            "height": 40,
            "marginLeft": 10,
            "marginTop": 10,
            "textAlign": "center",
            "textAlignVertical": "center",
            "width": 40,
          }
        }
      >
        4
      </Text>
    </View>
  </View>
  <View>
    <View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "red",
          }
        }
      >
        Score : 
        0
      </Text>
    </View>
  </View>
</View>
`;

exports[`renders the Welcome component 1`] = `
<View>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Awale
    </Text>
  </View>
  <View>
    <View
      style={
        Object {
          "marginBottom": 30,
        }
      }
    >
      <View
        accessibilityComponentType="button"
        accessibilityLabel={undefined}
        accessibilityTraits={
          Array [
            "button",
          ]
        }
        accessible={true}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#0C42FD",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            Play against computer
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "marginBottom": 30,
        }
      }
    >
      <View
        accessibilityComponentType="button"
        accessibilityLabel={undefined}
        accessibilityTraits={
          Array [
            "button",
          ]
        }
        accessible={true}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#0C42FD",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            Play against another player
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
